src/Model/Payment/Braintree/CreditCard.php

```diff
@@ -0,0 +1,231 @@
+
+            $extensionAttributes->setVaultPaymentToken($paymentToken);
+            $payment->setExtensionAttributes($extensionAttributes);
+        }
+        $payment->save();
+
+        $amount = $payment->formatAmount($shopgateOrder->getAmountComplete(), true);
+        $payment->setBaseAmountAuthorized($amount);
+        $payment->setShouldCloseParentTransaction(false);
+        $payment->setIsTransactionClosed(0);
+        $payment->registerAuthorizationNotification($shopgateOrder->getAmountComplete());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getAdditionalPaymentData(ShopgateOrder $shopgateOrder): array
+    {
+        $paymentInformation = $shopgateOrder->getPaymentInfos();
+
+        $additionalPaymentData = [
+            'method_title'               => $paymentInformation['shopgate_payment_name'],
+            'processorAuthorizationCode' => $paymentInformation['processor_auth_code'],
+            'processorResponseCode'      => $paymentInformation['processor_response_code'],
+            'processorResponseText'      => $paymentInformation['processor_response_text'],
+            'cc_number'                  => $this->helper->formatVisibleCCNumber(
+                $paymentInformation['credit_card']['masked_number']
+            ),
+            'cc_type'                    => $this->helper->formatVisibleCCType(
+                $this->getMappedCCType($paymentInformation['credit_card']['type'])
+            )
+        ];
+
+        if (!empty($paymentInformation['risk_data'])) {
+            $additionalPaymentData['riskDataDecision'] = $paymentInformation['risk_data']['decision'];
+            $additionalPaymentData['riskDataId']       = $paymentInformation['risk_data']['id'];
+        }
+
+        return $additionalPaymentData;
+    }
+
+    /**
+     * @param string $ccType
+     *
+     * @return string
+     */
+    private function getMappedCCType(string $ccType): string
+    {
+        return static::CREDIT_CARD_TYPE_MAP[$ccType];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function manipulateOrderWithPaymentDataAfterSave(
+        MagentoOrder $magentoOrder,
+        ShopgateOrder $shopgateOrder
+    ): void {
+        $payment = $magentoOrder->getPayment();
+        if ($shopgateOrder->getIsPaid()) {
+            // presumably Shopgate captured, so we should not
+            $payment->registerCaptureNotification($shopgateOrder->getAmountComplete(), true);
+        } elseif ($this->shouldCaptureOnline()) {
+            $payment->capture();
+        }
+    }
+
+    /**
+     * Checks of Braintree CC is configured to authorize and capture
+     *
+     * @return bool
+     */
+    private function shouldCaptureOnline(): bool
+    {
+        $configuredPaymentAction = $this->scopeConfig->getConfigByPath(self::XML_CONFIG_PAYMENT_ACTION)->getValue();
+
+        return $configuredPaymentAction === MethodInterface::ACTION_AUTHORIZE_CAPTURE;
+    }
+
+    /**
+     * @inheritDoc
```

Comment:
Add info for this guy as is to why it should be empty. It's not right away obvious and my first instinct is to remove this method.